icu4j/main/core/src/main/java/com/ibm/icu/text/DateFormatSymbols.java

```diff
@@ -1898,6 +1870,42 @@ private DateFormatSymbols(ULocale desiredLocale, ICUResourceBundle b, String cal
         initializeData(desiredLocale, b, calendarType);
     }
 
+    /**
+     * Convert era names map from CalendarSink to array, filling in missing values from fallback.
+     * @internal
+     * @deprecated This API is ICU internal only.
+     */
+    protected String[] initEras(String erasKey, Map<String, Map<String, String>> maps,
+            ICUResourceBundle calBundle, int maxEra) {
+        Map<String, String> eraNamesTable = maps.get(erasKey);
+        if (eraNamesTable == null) {
+            return null;
+        }
+        ICUResourceBundle calErasWidthBundle = calBundle.findWithFallback(erasKey);
+        String[] eraArray = new String[maxEra + 1];
+        if (eraArray != null) {
+            for (int eraCode = 0; eraCode <= maxEra; eraCode++) {
+                String eraKey = Integer.toString(eraCode);
+                String eraName = eraNamesTable.get(eraKey);
+                if (eraName != null) {
+                    eraArray[eraCode] = eraName;
+                } else {
+                    eraArray[eraCode] = "";
+                    if (calErasWidthBundle != null) {
+                        ICUResourceBundle calErasWidthKeyBundle = calErasWidthBundle.findWithFallback(eraKey);
+                        if (calErasWidthKeyBundle != null) {
+                            eraName = calErasWidthKeyBundle.getString();
```

Comment:
Why do we need to do a whole separate lookup here?
I don't know how the `maps` are collected, but it seems like having the sink read the data should assemble all of the values from the parent bundles into the same eraNamesTable. If not, then why is that not happening?
Is this just filling in era names of a different width?
If so, then a comment would help.

Reply:
@markusicu For a map, the sink does not seem to fill in parent entries for keys that do not exist in the current bundle (I guess it has no way to know what those keys should be be), that is why we need to explicitly fill these in. Also true in ICU4C. I can add a comment.

Reply:
When the sink is called at any one level, it does not have access to the data at other levels.
For a table, the sink should collect the data into a Map (or similar), and add data at each level (as it descends to the parents) if the key is still absent.
Once the sink loading is done, the Map should have a full set of entries from all bundles.

Reply:
@markusicu Probably (though I do not see an obvious issue there), but maybe we can do that separately in a later PR so this PR can unblock the changes Shane wants to make (and which will need to be integrated). I can leave the ticket open for a later PR.